File: problems/python/stock_profits.py
"""121. Best Time to Buy and Sell Stock"""

from typing import List


class Solution:
    def maxProfit(self, prices: List[int]) -> int:
        sell = 0
        buy = prices[0]

        for p in prices:
            profit = p - buy
            if profit > sell:
                sell = profit
            if p < buy:
                buy = p

        return sell

File: problems/typescript/stock_profits.ts
/*
* 121. Best Time to Buy and Sell Stock
*/

function maxProfit(prices: number[]): number {
    let high: number = Math.max();
    let low: number = prices[0];
    for (let i = 0; i < prices.length; i++) {
        let profit: number = prices[i] - low;
        if (profit > high) {
            high = profit;
        }
        if (prices[i] < low) {
            low = prices[i];
        }
    }
    return high;
};